tests/migrations/test_uppercase_app_name.py

```diff
@@ -0,0 +1,15 @@
+from django.db.migrations.operations.utils import resolve_relation
+from django.test import SimpleTestCase
+
+
+class MigrationsUppercaseAppNameTests(SimpleTestCase):
+
+    def test_resolve_relation(self):
+        """
+        Ensure that the resolve_relation() function keeps uppercase
+        letters in the app name intact.
+        """
+        APP_NAME = "Uppercase_AppName"
+        MODEL_NAME = "ModelName"
+        app_name, model_name = resolve_relation(APP_NAME + "." + MODEL_NAME)
+        self.assertEqual(app_name, APP_NAME)
```

Comment:
We don't need a separate file and class for a single test. Please move it to the `tests/migrations/test_autodetector.py` and add test with a real scenario i.e. create a minimal model state and check changes detected by `MigrationAutodetector`.

Reply:
OK, I have added a test with a real scenario in 9a5ce08